### PR TITLE
Always dump cluster state

### PR DESCRIPTION
**Pull request by crusaderky**
Dump cluster state for all successful runs and dump URI to the database.

### CODE REVIEW

File: benchmark_schema.py [hunk 1]
```diff
@@ -26,6 +26,8 @@ class TestRun(Base):
     distributed_version = Column(String, nullable=True)
     python_version = Column(String, nullable=True)
     platform = Column(String, nullable=True)
+    cluster_name = Column(String, nullable=True)
+    cluster_id = Column(Integer, nullable=True)
```
Comment (ian-r-rose) [inline benchmark_schema.py:30]: Not that @ncclementi is doing the same thing in #401. Also, this will require an alembic migration

Comment (ncclementi) [reply benchmark_schema.py:30]: PR for reference, https://github.com/coiled/coiled-runtime/pull/433

Comment (crusaderky) [reply benchmark_schema.py]: I've reverted my version

**Comment by crusaderky** [comment]
image

Tested I can download the file by simply typing `aws s3 cp <paste from db> ./`

**Comment by crusaderky** [comment]
This is ready for review and merge

File: benchmark_schema.py [hunk 2]
```diff
@@ -39,12 +39,12 @@ class TestRun(Base):
     average_memory = Column(Float, nullable=True)
     peak_memory = Column(Float, nullable=True)
 
-    # Timing data (Note: this is not yet collected)
+    # Timing data (not yet collected)
```
Comment (ian-r-rose) [inline benchmark_schema.py:42]: Hah, this comment is out of date, we are collecting this info now. But I can update separately.

File: conftest.py [hunk 3]
```diff
@@ -434,7 +432,7 @@ def small_client(
         client.wait_for_workers(10)
         client.restart()
 
-        with upload_cluster_dump(client, small_cluster), benchmark_all(client):
```
Comment (ian-r-rose) [inline conftest.py:437]: Thanks, I've been meaning to make this change.

File: dashboard.py [hunk 4]
```diff
@@ -242,9 +242,13 @@ def make_ab_confidence_map(
             "runtime",
         ]
     ]
+
     runtimes = df["runtime"].unique()
+    if len(runtimes) < 2 or baseline not in runtimes:
```
Comment (ian-r-rose) [inline dashboard.py:247]: I've made a similar fix in #432. But will this actually solve the problem? I think the problem that I saw was when `runtimes` was entirely full of `baseline`, not if `baseline` didn't appear.

Comment (crusaderky) [reply dashboard.py]: I now moved this fix to #434 and it does solve the problem

**Comment by crusaderky** [comment]
@ian-r-rose is there anything outstanding here?

**Review by ian-r-rose — APPROVED** [review]
LGTM, thanks @crusaderky

**Pull request merged**